wire-java-generator/src/main/java/com/squareup/wire/java/JavaGenerator.java

```diff
@@ -423,7 +423,22 @@ private static void putAllExtensions(
         if (!eligibleAsAnnotationMember(schema, field)) continue;
 
         ProtoMember protoMember = extend.member(field);
-        memberToJavaName.put(protoMember, annotationName(protoFile, field, new ClassNameFactory()));
+        ClassName annotationName =
+            annotationName(protoFile, field, new ClassNameFactory(), "Option");
+        if (memberToJavaName.containsValue(annotationName)) {
+          // To avoid conflicts for same named options on different types, we generate a more
+          // precise name. i.e. 'ObjectiveOption' will become 'ObjectiveFieldOption'.
+          String extendSimpleName = extend.getType().getSimpleName();
+          memberToJavaName.put(
+              protoMember,
+              annotationName(
+                  protoFile,
+                  field,
+                  new ClassNameFactory(),
+                  extendSimpleName.substring(0, extendSimpleName.length() - 1)));
```

Comment:
Will `extendSimpleName` here always have the value `FieldOptions`, or it can also have the value `MessageOptions`?

Reply:
Can be any option type, `FieldOption`, `MessageOption`, `EnumValueOption`, etc. This is how we avoid the conflict